pennylane/beta/plugins/default_tensor.py

```diff
@@ -355,41 +366,74 @@ class DefaultTensor(Device):
     C_DTYPE = np.complex128
     R_DTYPE = np.float64
 
-    def __init__(self, wires, shots=1000, analytic=True):
+    zero_state = np.array([1.0, 0.0], dtype=C_DTYPE)
+
+    def __init__(self, wires, shots=1000, analytic=True, representation="exact"):
         super().__init__(wires, shots)
-        self.analytic = True
-        self._nodes = []
-        self._edges = []
-        self._state_node = None
-        self._free_edges = []
+        self.analytic = analytic
+        self._rep = representation
         self.reset()
 
-    @staticmethod
-    def _create_basis_state(state, wires):
-        """Helper function to create a basis state with the correct shape.
+    def reset(self):
+        """Reset the device"""
+        self._clear_network()
 
-        Args:
-            state (array[int]): array of 0s and 1s of size ``(wires,)`` representing
-                the basis state
-            wires (list[int]): the wires the basis state should
-                be prepared on
+        # prepare a factorized all-zeros state
+        self._add_initial_state_nodes(
+            [self.zero_state] * self.num_wires,
+            [[w] for w in range(self.num_wires)],
+            ["ZeroState"] * self.num_wires,
+        )
 
-        Returns:
-            array[int]: state array of size ``[2]*len(wires)``
+    def _clear_network(self):
+        """Remove all data representing the current network from internal cache."""
+        self._nodes = {}
+        if self._rep == "exact":
+            self._contracted = False
+            self._terminal_edges = []
+
+        elif self._rep == "mps":
+            raise NotImplementedError
+            # nodes = []
+            # for w in range(self.num_wires):
+            #    tensor = np.reshape(self.zero_state, [1, 2, 1])  # this shape is required, even for end nodes it seems
+            #    node = self._add_node(tensor, wires=[w], name="ZeroState")
+            #    nodes.append(node)
+            #    if w > 0:
+            #        tn.connect(nodes[w-1][2], nodes[w][0])
+            ## Note: might want to set canonicalize=False
+            # self.mps = tn.matrixproductstates.finite_mps.FiniteMPS(nodes)
+
+    def _add_initial_state_nodes(self, tensors, wires, names):
+        """Create the nodes representing the initial input state circuit.
+        
+           Input states can be factorized or entangled. If a state can be factorized
+           into :math:`k` subsystems, then ``tensors``, ``wires``, and ``names`` should be lists of length :math:`k`.
+
+          Args:
+              tensors (Sequence[np.array, tf.Tensor, torch.Tensor]): the numerical tensors for each factorized component of
```

Comment:
Curious here: the types imply that this would be inherited in `default.tensor.tf`, correct? Does it also work with `torch.Tensor`s? (Perhaps that gets converted by `tensornetwork`? :thinking: )

Reply:
yep, tensornetwork doesn't care what type of data tensors you use

Reply:
In fact, other than `_operation_map` and `_observable_map`, nothing in this device is really specific to the type of tensor you use. Should work be default for numpy, jax, tensorflow, and pytorch (a feature of tensornetwork)

Reply:
Oh, that's something really valuable to know, thank you!

So if we compare using:
1. `default.tensor` with `tf.Tensor` objects
2. `default.tensor.tf` (with `tf.Tensor` objects)

then the difference seems to be
1. option: uses `numpy` operations and the specified backend to compute gradients
2. option: uses `tf` operations and `tf.GradientTape` to compute the gradient

Would that be fair to say that these are the differences (and the performance) or would there be something else as well?

Reply:
Yep, that sums it up @antalszava! As a result of `default.tensor.tf` using `tf` operations and `tf.GradientTape` to compute the gradient, it also supports being used as a `PassthruQNode` and a device jacobian QNode.